### DISCUSSION TITLE
Generation of the negative CPRED data

### DISCUSSION THREAD

**Discussion by ckcho0625** [nlmixr2/nlmixr2#213] [Q&A]
Dear Fidler and nlmixr2 users

I have a one question for the generated CPRED data 

I am currently building a population pharmacokinetic model using nlmixr2. Unexpected results have been observed, where CPRED values are predicted as negative, resulting in a goodness-of-fit plot that shows a negative slope. Despite testing various models, some CPRED values and slope of DV vs. CPRED have consistently been predicted as negative.  The generated CPRED data is quite different to the generated PRED data. When I tested the model with other data I have, normal results (positive DV vs. CPRED slope) were observed. I would like to inquire whether there might be an issue with the plasma concentration data I have. I wonder how can I solve this problem. I attached the code and the goodness-of-fit profiles.

Kind regards

![124d4d55-e1b7-411c-9ff8-6391c3578240](https://github.com/nlmixr2/nlmixr2/assets/63388711/3c73c68a-673a-47b8-aa8c-087c8d8defc1)

![9e419cb9-3e20-4509-90ad-ce48f92fb5f1](https://github.com/nlmixr2/nlmixr2/assets/63388711/79abff86-a3bd-465e-8848-4b540e2df339)

Model code
  practice <- function() {
    ini({
      TCL <- log(2944.2); label("CL")       # 2944.2: calculated CL/F of NCA (L/h)
      TV  <- log(7833.5); label("V")         # 7833.5: calculated Vz/F of NCA (L)
      TKA <- log(0.5); label("Ka")     

      Eta.CL + Eta.V ~ c(1.0, 1.0, 1)
      Eta.KA ~ 1.0
      prop.err <- 1.0
    })
    model({
      CL <-  exp(TCL+Eta.CL)
      V  <- exp(TV+Eta.V)
      KA <- exp(TKA+Eta.KA)
      d/dt(depot) = -KA*(depot)
      d/dt(cent) = KA*(depot)-CL/V*(cent)
      Cp = cent/V
      Cp ~ prop(prop.err)
    })
  }

nlme_practice <- nlmixr2(practice, PKdata, "saem", table=tableControl(cwres=T,npde = T))
print(nlme_practice)

**Comment by billdenney** [comment]
The way that you have specified your model's inter individual variability structure, it will be numerically unstable.  You have specified perfectly correlated IIV between CL and V (`Eta.CL + Eta.V ~ c(1.0, 1.0, 1)`).  I would guess that if you specified a more typical correlation, `Eta.CL + Eta.V ~ c(1.0, 0.2, 1)`, it will likely work as you expect.

**Comment by mattfidler** [comment]
One other thing to keep in mind is you are using `saem` which is not the focei approximation.  The solution that you have may violate the assumptions of the focei approximation since the `saem` estimation routine does not use it.

**Reply by mattfidler** [reply]
Perhaps in the case of `saem`, the `npd` is the more appropriate metric (note `npde` may introduce spurious correlations).  Monolix does not generate `cwres` in general partially because they believe `npd` is a better metric for goodness of fit in `saem`.

**Comment by mattfidler** [comment]
This seems like more of a disucssion so I am moving it to the discussions so it will be easier for others to find in the future.

**Comment by ckcho0625** [comment]
Thank you for your responses. However, I have tried various methods, including those you suggested (using FOCEI estimation and adjusting Eta values), but in most cases, we still encounter situations where negative values are estimated. I have found that CPRED becomes positive only when specific conditions are applied to initial values and Eta values, yet the final estimated values differ significantly from the NCA results. As my understanding of population pharmacokinetics, including nlmixr2, is still somewhat limited, I kindly request further advice to resolve this issue.

Kind regards

**Reply by mattfidler** [reply]
Hi @ckcho0625 

I would say that even though I use `CPRED` as a default prediction, it does not ensure that it is positive.  You can see in this publication:

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5321813/pdf/PSP4-6-87.pdf

You can try to look at other metrics like `EPRED`, or `PRED` to see if these seem reasonable.  To be frank I am unsure what a negative `CPRED` indicates.  I have also seen it in some of my models in the past, but `EPRED` `PRED` give reasonable values.

As a note, they also suggest looking at `PPRED` which actually is the same as `EPRED` in nlmixr2.  (The outputs match NONMEM instead of other tools which may call this `PPRED`)

**Reply by mattfidler** [reply]
In my opinion, `EPRED` is a more robust metric than `CPRED`, and you can see what you want to see with `CWRES` vs `EPRED` or `NPD` vs `EPRED`

**Reply by mattfidler** [reply]
Also looking at the equation, it may be that your `eta` values for certain subjects are quite big;  Maybe you can see if the large `etas` correspond to negative `CPRED` values.
